module/zfs/spa_misc.c

```diff
@@ -1784,7 +1785,9 @@ uint64_t
 spa_get_slop_space(spa_t *spa)
 {
 	uint64_t space = spa_get_dspace(spa);
-	return (MAX(space >> spa_slop_shift, MIN(space >> 1, spa_min_slop)));
+	return (MIN(spa_max_slop,
+	    MAX(space >> spa_slop_shift,
+	    MIN(space >> 1, spa_min_slop))));
```

Comment:
It looks like the comment above this function needs to be updated too.

Reply:
Yea, there's a few comments that'll need to be updated. I didn't worry about that yet, until we all think this is something we can actually move forward.

Reply:
I think we can and should.  128GiB feels plenty conservative.

Reply:
@ahrens do you agree with @behlendorf?

I was going to try and see how I might gather some data to verify if this is a reasonable upper bound, but I'm not really sure how to do that, and didn't make much progress toward that goal during the hackathon yesterday.